maestro-studio/web/src/InteractPage.tsx

```diff
@@ -0,0 +1,43 @@
+import InteractableDevice from './InteractableDevice';
+import ReplView from './ReplView';
+import React, { useEffect, useState } from 'react';
+import { DeviceScreen } from './models';
+import { API, wait } from './api';
+
+const InteractPage = () => {
+  const [deviceScreen, setDeviceScreen] = useState<DeviceScreen>()
+
+  useEffect(() => {
+    let running = true;
+    (async () => {
+      while (running) {
+        try {
+          const deviceScreen = await API.getDeviceScreen(true)
+          setDeviceScreen(deviceScreen)
+        } catch (e) {
+          console.error(e)
+          await wait(1000)
+        }
+      }
+    })()
+    return () => { running = false }
+  }, [setDeviceScreen])
```

Comment:
Should this be an interval instead of a blocking while loop?

Reply:
This isn't blocking actually, it using a async wait function that uses timeout under the hood. This is essentially what third party interval hooks use (I don't know of a built-in hook that does this)

Reply:
Ah sorry I meant within the async anonymous function that essentially handles the ticking of the interval. It just seems a bit odd to me to not use `setInterval` and then `clearInterval` upon unmounting and instead handle this manually but not a big deal so I'm fine to leave as is 👍

Reply:
Ah ok - I thought about setInterval, but the difference is that setInterval runs a function on a set time interval whereas this logic refreshes the device screen sequentially as frequently as possible. If used setInterval, (1) we wouldn't be refreshing as quickly as possible and/or (2) we'd encounter race conditions with simultaneous async http requests.